syntax = "proto3";
package cmdb_service_ctrl;




import "google/protobuf/struct.proto";

/*
SearchSubSystemAll请求
*/
message SearchSubSystemAllRequest {
    /*
    系统的实例ID
    */
    string systemInstanceId = 1 ;
    /*
    查询条件,与[实例搜索接口]的query字段说明一致
    */
    google.protobuf.Struct query = 2 ;
    /*
    过滤的字段列表, 留空代表返回所有字段(true: 表示指定获取字段, false: 表示指定不获取的字段)(支持关系数据的二级jsonPath格式的指定字段如clusters.name)
    */
    google.protobuf.Struct fields = 3 ;
    /*
    按字段排序, 留空默认按照实例ID降序排序(1表示升序, -1表示降序)
    */
    google.protobuf.Struct sort = 4 ;
    /*
    按照权限过滤(通用实例都有`read`, `update`, `delete`权限控制, 主机实例在通用实例权限基础上有额外的`operate`权限, 应用实例在通用实例权限基础上有额外的`developClusterOperate`, `testClusterOperate`, `prereleaseClusterOperate`, `productionClusterOperate`权限)
    */
    repeated string permission = 5 ;
    /*
    对于关联的实例数据是否只获取relation_view中指定的属性, 这个字段为true时, 会覆盖fields字段中指定的二级字段设置
    */
    bool only_relation_view = 6 ;
    /*
    是否只获取与自己有关的那部分数据, 默认为false
    */
    bool only_my_instance = 7 ;
}

/*
SearchSubSystemAll返回
*/
message SearchSubSystemAllResponse {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    错误信息
    */
    string error = 2 ;
    /*
    返回消息
    */
    string message = 3 ;
    /*
    实例列表
    */
    repeated google.protobuf.Struct data = 4 ;
}

/*
SearchSubSystemAllApi返回
*/
message SearchSubSystemAllResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    cmdb_service_ctrl.SearchSubSystemAllResponse data = 4 ;
}

